pkg/shared/utils/arrays/generic_array_utils.go

```diff
@@ -99,3 +99,8 @@ func ForEach[T any](values []T, consumer ConsumerFunc[T]) {
 		consumer(values[i])
 	}
 }
+
+// IsEmpty returns true if the array is empty, otherwise returns false.
```

Comment:
Can we clarify that an array is considered "empty" in this context if the array is nil or if the array is not nil but has zero elements?

Reply:
Thanks, i'll add that.

Reply:
done